src/cdk/v2/destinations/pinterest_tag/procWorkflow.yaml

```
@@ -65,6 +74,7 @@ steps:
         "language": .properties.language ??
                     .context.traits.language ??
                     .context.locale,
+        "advertiser_tracking_enabled": $.outputs.apiVersion === {{$.API_VERSION.v5}} && $.outputs.isAdvertisementMappingEnabled && .context.device.adTrackingEnabled !== undefined ? Boolean(.context.device.adTrackingEnabled),
```

Wouldn't this be sufficient? Moreover, I wouldn't coerce the payload value into a boolean.

```suggestion
        "advertiser_tracking_enabled": $.outputs.apiVersion === {{$.API_VERSION.v5}} && $.outputs.isAdvertisementMappingEnabled && .context.device.adTrackingEnabled === true),
```

---

This is not we want, it will leads to scenario where adTrackingEnabled comes as undefined it will be treated as false

---

I will change to 
```
"advertiser_tracking_enabled": $.outputs.apiVersion === {{$.API_VERSION.v5}} && $.outputs.isAdvertisingMappingEnabled && .context.device.adTrackingEnabled !== undefined ? Boolean(.context.device.adTrackingEnabled) : undefined,
```

---

If we set undefined, it will be removed by [removeUndefindValues](https://github.com/rudderlabs/rudder-transformer/pull/4733/files#diff-7a4dcbf4a044d1f82480d9330dfc15b594483f8cbcc551d3bb6ee66072ebc7d9R87) while making final payload